src/couch_index/src/couch_index_server.erl

```diff
@@ -286,67 +302,92 @@ reset_indexes(DbName, #st{} = State) ->
 add_to_ets(DbName, Sig, DDocId, Pid, #st{} = St) ->
     ets:insert(St#st.by_sig, {{DbName, Sig}, Pid}),
     ets:insert(St#st.by_pid, {Pid, {DbName, Sig}}),
-    ets:insert(St#st.by_db, {DbName, {DDocId, Sig}}).
+    ets:insert(St#st.by_db, {DbName, {DDocId, Sig}}),
+    ets:delete(St#st.openers, Pid).
```

Comment:
I think you're confusing pids here. The openers contain opener pids, where this Pid is an indexer pid that's not in openers. Check the `new_index` code to show how we send back the indexer pid before unlinking it.